xarray/core/indexing.py

```diff
@@ -371,13 +465,21 @@ def shape(self):
         return tuple(shape)
 
     def __array__(self, dtype=None):
-        array = orthogonally_indexable(self.array)
+        array = broadcasted_indexable(self.array)
         return np.asarray(array[self.key], dtype=None)
 
     def __getitem__(self, key):
+        if isinstance(key, VectorizedIndexer):
+            raise NotImplementedError('Vectorized indexing for {} is not '
+                                      'implemented.'.format(type(self)))
+        key = expanded_indexer(key, self.ndim)
```

Comment:
I would suggest putting these four lines inside `_updated_key()`, since they're repeated here and in `__setitem__`